packages/components/RadioGroup/src/RadioButton.macos.tsx

```diff
@@ -0,0 +1,56 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { ensureNativeComponent } from '@fluentui-react-native/component-cache';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { RadioGroupContext } from './RadioGroup';
+import { radioButtonName, IRadioButtonProps, IRadioButtonSlotProps, IRadioButtonType } from './RadioButton.types';
+
+const NativeRadioButtonView = ensureNativeComponent('RadioButtonView');
+
+export const RadioButton = compose<IRadioButtonType>({
+  displayName: radioButtonName,
+  usePrepareProps: (userProps: IRadioButtonProps, useStyling: IUseComposeStyling<IRadioButtonType>) => {
+    const { content, buttonKey, disabled } = userProps;
+    const info = React.useContext(RadioGroupContext);
+    // Reroute the native component's onPress event to RadioGroup's onChange
+    const onPressRerouted = () => {
+      // Prevent calls to RadioGroup's onChange on the currently selected button
+      if (buttonKey != info.selectedKey) {
+        info.onChange && info.onChange(buttonKey);
+      }
+    };
+
+    const styleProps = useStyling(userProps);
+    const slotProps = mergeSettings<IRadioButtonSlotProps>(styleProps, {
+      root: {
+        buttonKey: buttonKey,
+        content: content,
+        disabled: disabled,
+        onPress: onPressRerouted,
+        selected: info.selectedKey === buttonKey,
```

Comment:
nit, I would move the equality check to it's own line for readability, so that `const slotProps = ...` has less logic inside of it